Runtime/Reporters/SlackMessageSender.cs

```diff
@@ -0,0 +1,154 @@
+    /// delegate touching Slack API to <c cref="SlackAPI" />.
+    /// Purpose of introducing this interface is making <c cref="SlackReporter" /> be a humble object.
+    /// </summary>
+    public interface ISlackMessageSender
+    {
+        /// <summary>
+        /// Post report log message, stacktrace and screenshot
+        /// </summary>
+        /// <param name="slackToken">Slack API token</param>
+        /// <param name="slackChannel">Slack Channel to send notification</param>
+        /// <param name="mentionSubTeamIDs">Sub team IDs to mention</param>
+        /// <param name="addHereInSlackMessage">Whether adding @here or not</param>
+        /// <param name="logString">Log message</param>
+        /// <param name="stackTrace">Stack trace</param>
+        /// <param name="withScreenshot">With screenshot</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns></returns>
+        UniTask Send(
+            string slackToken,
+            string slackChannel,
+            IEnumerable<string> mentionSubTeamIDs,
+            bool addHereInSlackMessage,
+            string logString,
+            string stackTrace,
+            bool withScreenshot,
+            CancellationToken cancellationToken = default
+        );
+    }
+
+    /// <summary>
+    /// A class for Slack message senders. This class defines a message format and delegates touching Slack API to
+    /// <c cref="SlackAPI" />.
+    /// </summary>
+    public class SlackMessageSender : ISlackMessageSender
+    {
+        private readonly SlackAPI _slackAPI;
+        private readonly string _slackToken;
+
+
+        /// <summary>
+        /// Creates a new instance for <c cref="SlackMessageSender" />.
+        /// </summary>
+        /// <param name="api">Slack API client</param>
+        public SlackMessageSender(SlackAPI api)
+        {
+            _slackAPI = api;
+        }
+
+
+        /// <inheritdoc />
+        public async UniTask Send(
+            string slackToken,
+            string slackChannel,
+            IEnumerable<string> mentionSubTeamIDs,
+            bool addHereInSlackMessage,
+            string logString,
+            string stackTrace,
+            bool withScreenshot,
+            CancellationToken cancellationToken = default
+        )
+        {
+            if (string.IsNullOrEmpty(slackToken) || string.IsNullOrEmpty(slackChannel))
+            {
+                return;
+            }
+
+            var title = Title(logString, mentionSubTeamIDs, addHereInSlackMessage);
+
+            await UniTask.SwitchToMainThread();
+
+            var postTitleTask = await _slackAPI.Post(
+                slackToken,
+                slackChannel,
+                title,
+                cancellationToken: cancellationToken
+            );
+            if (!postTitleTask.Success)
+            {
+                return;
+            }
+
+            if (withScreenshot && !Application.isBatchMode)
```

Comment:
IMO
There is a workaround for the batchmode problems.bellow:
```
using TestHelper.RuntimeInternals.Wrappers.UnityEditor

GameViewWrapper.GetWindow();
```

see: https://www.nowsprinting.com/entry/2023/06/14/003659

Reply:
I'm gonna try to use you suggested on another PR.